### PR TITLE
page: Move content under docs and create initial site

### PR DESCRIPTION
**Pull request by qinqon**
**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
> /kind enhancement

**What this PR does / why we need it**:
Create a initial webpage with the current content

**Special notes for your reviewer**:
Depends on https://github.com/kubevirt/project-infra/pull/698

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by qinqon** [comment]
> Use symlinks or copy for the logo if you need. The repository should keep our logo directory visible, so we know where to find it. It is not meant only as a documentation asset.

@phoracek what's the problem on having it at `docs/assets/images/logo`, it's not hidden.

**Comment by qinqon** [comment]
/hold cancel

We have preview now

File: README.md [hunk 1]
```diff
@@ -1,6 +1,4 @@
-# kubernetes-nmstate
```
Comment (phoracek) [inline README.md:1]: Keep the header, it is still valuable for people reading plaintext README.md

File: README.md [hunk 2]
```diff
@@ -1,6 +1,4 @@
...
-
-<img src="logo/fullcolor.png" alias="project logo" />
+<img src="docs/assets/images/logo/fullcolor.png" alias="project logo" />
```
Comment (phoracek) [inline README.md:1]: Logos should stay under `logo/`. They are assets of the project, not of the documentation. Do not hide them down there.

File: docs/_config.yaml [hunk 3]
```diff
@@ -1,3 +1,45 @@
-# _config.yml
-
-title: "Jekyll Actions Demo"
+theme: minimal-mistakes-jekyll
+minimal_mistakes_skin: "air"
+locale: "en-US"
+title: "The Kubernetes NMState project"
+description: A Declarative API for Kubernetes Node Network Management
```
Comment (phoracek) [inline docs/_config.yaml:5]: Since this is just a description and not a title, it should use normal capitalization:

A declarative API for Kubernetes node network management

File: docs/_config.yaml [hunk 4]
```diff
@@ -1,3 +1,45 @@
...
+name: "The Kubernetes nmstate team"
```
Comment (phoracek) [inline docs/_config.yaml:6]: NMState

File: docs/_pages/examples.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+---
+title: Examples
+layout: collection
+permalink: /examples/
+collection: examples
+entries_layout: grid
```
Comment (phoracek) [inline docs/_pages/examples.md:6]: I prefer the single page with a list of examples for now. It requires less clicking through for users. With the sidebar listing all examples it is also quite easy to navigate.

Eventually I hope to have detailed docs about topics covering multiple examples and keep them with description. For now though, I'd go with single page.

**Review by phoracek — CHANGES_REQUESTED** [review]
Links in the page header on 3 different sections don't lead anywhere

File: README.md (hunk 2 above)
Comment (qinqon) [reply README.md:1]: I will create a link from docs to logo doing the opposite is not working and I prefer not to duplicate stuff.

Comment (qinqon) [reply README.md]: Symlinks not working with markdown

**Comment by qinqon** [comment]
/hold

examples description missing, we have to fix the logo issue.

File: README.md [hunk 6]
```diff
@@ -1,6 +1,5 @@
-# kubernetes-nmstate
-
 <img src="logo/fullcolor.png" alias="project logo" />
+# kubernetes-nmstate
```
Comment (phoracek) [inline README.md:2]: Since there is no space above this, this does not render as a title. Please move it on the top of the file as it was and maintain the empty line

File: docs/_pages/user-guide-102-configuration.md [hunk 7]
```diff
@@ -26,27 +30,10 @@ First of all, let's apply the manifest:
...
-      state: up
-      ipv4:
-        dhcp: true
-        enabled: true
-      link-aggregation:
-        mode: balance-rr
-        slaves:
-        - eth1
-        - eth2
+{% include_absolute 'user-guide/bond0-eth1-eth2_up.yaml' %}
```
Comment (phoracek) [inline docs/_pages/user-guide-102-configuration.md:36]: Oh how cool, testable user-guide!

File: docs/_pages/deployment-local-cluster.md [hunk 8]
```diff
@@ -1,4 +1,8 @@
-# Deployment on Local Cluster
+---
```
Comment (phoracek) [inline docs/_pages/deployment-local-cluster.md:1]: This article contains markdown links which lead to 404.

**Review by phoracek — CHANGES_REQUESTED** [review]

File: docs/_pages/user-guide-101-reporting.md [hunk 9]
```diff
@@ -187,4 +191,4 @@ The `NodeNetworkState` will now list all interfaces seen on the host.
 ## Continue reading
 
 The following tutorial will guide you through the configuration of node
-networking: [Configuration](user-guide-102-configuration.md)
+networking: [Configuration](user-guide/102-configuration)
```
Comment (phoracek) [inline docs/_pages/user-guide-101-reporting.md:194]: This leads to userguide/101-configuration/userguide/102-configuration

**Review by phoracek — CHANGES_REQUESTED** [review]

File: docs/_pages/deployment.md [hunk 10]
```diff
@@ -0,0 +1,8 @@
+---
+title: Deployment
+permalink: /deployment/
+---
+{% assign deployment_pages = site.pages | where: "tag","deployment" %}
+{% for page in deployment_pages %}
+  [ {{ page.title }} ]({{ site.baseurl }}{{ page.url }})
```
Comment (phoracek) [inline docs/_pages/deployment.md:7]: These links do not work

**Review by phoracek — CHANGES_REQUESTED** [review]
Found more links that don't work. Have not commented on all of them. Would you please go through the page and check all? Or add a sanity check making sure that we don't have dead links.

**Comment by qinqon** [comment]
/hold 
Add sanity checker for links

**Comment by qinqon** [comment]
links should be fixed now and new job `
pull-kubernetes-nmstate-docs ` is checking them.

**Review by rhrazdil — COMMENTED** [review]
There seems to be one comment from Petr that wasn't addressed, otherwise it looks good to me

File: README.md (hunk 6 above)
Comment (phoracek) [reply README.md:2]: @qinqon ^^

File: docs/_plugins/include_absolute.rb [hunk 11]
```diff
@@ -0,0 +1,168 @@
+module Jekyll
```
Comment (phoracek) [inline docs/_plugins/include_absolute.rb:1]: Is this module taken from somewhere or is it our doing?

File: docs/Makefile [hunk 12]
```diff
@@ -1,14 +1,31 @@
...
 	bundle update
 
-s serve:
-	bundle exec jekyll serve --source sample_site --destination build/ --livereload --trace
+.PHONY: build
+build:
+	rm -rf $(dest_dir)
+	bundle exec jekyll build --trace --source . --destination $(dest_dir)
+
+.PHONY: check
```
Comment (phoracek) [inline docs/Makefile:24]: bad phony

File: docs/Dockerfile [hunk 13]
```diff
@@ -0,0 +1,6 @@
+ARG  BRANCH=master
+FROM quay.io/nmstate/kubernetes-nmstate-docs:${BRANCH}
+COPY Makefile Gemfile* /docs/
+RUN make -C docs install
+COPY . /docs/
+RUN make -C docs check
```
Comment (phoracek) [inline docs/Dockerfile:6]: We don't have check anymore. And why do we call it while building the container?

File: README.md (hunk 6 above)
Comment (qinqon) [reply README.md:2]: Ahh, sorry

Comment (qinqon) [reply README.md:2]: Done

File: docs/.dockerignore [hunk 14]
```diff
@@ -0,0 +1,12 @@
+# We cannot call `docker build` out of top dir since this is controlled by
```
Comment (phoracek) [inline docs/.dockerignore:1]: I don't get how is this related to this dockerfile

File: automation/publish.sh [hunk 15]
```diff
@@ -6,9 +6,34 @@
...
+)
+
+# Update kubernetes-nmstate documentation container
+docs_container=${image_registry}/${image_repo}/kubernetes-nmstate-docs:master
+docker build docs --build-arg BRANCH=master -t $docs_container
+docker push $docs_container
+
+# To pass user/password from automations, idea is taken from [1]
+# [1] https://stackoverflow.com/questions/8536732/can-i-hold-git-credentials-in-environment-variables
+git config credential.helper '!f() { sleep 1; echo "username=${GITHUB_USER}"; echo "password=${GITHUB_TOKEN}"; }; f'
```
Comment (phoracek) [inline automation/publish.sh:29]: Would this hide our secrets? Would you set `set +x` to make extra sure to hide it?

**Review by phoracek — CHANGES_REQUESTED** [review]

File: automation/check-patch.docs.sh [hunk 16]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash -xe
+
+# Change url to point to google storage
+url="https://storage.googleapis.com"
```
[the remark this answers is not in the document]
Comment (qinqon) [reply automation/check-patch.docs.sh:4]: No clue this is the same places as other artifacts from jobs.

File: docs/_plugins/include_absolute.rb (hunk 11 above)
Comment (qinqon) [reply docs/_plugins/include_absolute.rb:1]: It's taken

File: automation/check-patch.docs.sh (hunk 16 above)
[the remark this answers is not in the document]
Comment (phoracek) [reply automation/check-patch.docs.sh:4]: I'm dumb! You are not uploading it, just seding.

File: docs/Makefile (hunk 12 above)
Comment (qinqon) [reply docs/Makefile:24]: the name of the Makefile target is `check` I think .PHONY is right.

File: docs/Dockerfile (hunk 13 above)
Comment (qinqon) [reply docs/Dockerfile:6]: We do have `check` target and depends on `build`, Dockerfile is consumed by the prow automation that checks links are broken and the like with `html-proofer`

File: docs/Makefile (hunk 12 above)
Comment (phoracek) [reply docs/Makefile:24]: I'm sorry. I'm blind.

File: docs/Dockerfile (hunk 13 above)
Comment (phoracek) [reply docs/Dockerfile:6]: Usually, Dockerfile is used to build a container image containing needed tools and only then the build image is used to use the tool.

Is not this the case here? Are we making the check as a part of the container building?

File: docs/.dockerignore (hunk 14 above)
Comment (qinqon) [reply docs/.dockerignore:1]: Will remove the comment and add sections per ignore.

File: automation/publish.sh (hunk 15 above)
Comment (qinqon) [reply automation/publish.sh:29]: We have the same thing here, I think the password does not appear https://github.com/nmstate/kubernetes-nmstate/blob/master/automation/release.sh#L10.

File: docs/Dockerfile (hunk 13 above)
Comment (qinqon) [reply docs/Dockerfile:6]: This docker file contains the build and the webpage, and is based on the same container published so we do incremental builds.

I can move the webpage build out of the Dockerfile but I think it does not worth it.

One of the cool things is that you can start the page without the repo just `docker run ... make serve`

**Comment by qinqon** [comment]
@phoracek can you re-review ?

File: automation/publish.sh [hunk 17]
```diff
@@ -6,9 +6,40 @@
...
+        push-handler
+)
+
+if [ "$branch" != $docs_source_branch ]; then
+    exit 0
+fi
+
+# Update kubernetes-nmstate documentation container
+docker build docs --build-arg BRANCH=$docs_source_branch -t $docs_container
+docker push $docs_container
```
Comment (phoracek) [inline automation/publish.sh:31]: Why is it important to push it? Is it just to cache it for users? And if so, is pulling of this container faster than building it locally? Is it worth the effort? My worry with builder images fetched from the remote registry is that if you need to change something in them it can be quite annoying.

Won't fight against this though.

File: automation/publish.sh [hunk 18]
```diff
@@ -6,9 +6,40 @@
...
-    push-handler
+(
+    cd ${TMP_PROJECT_PATH}
+    make \
+        IMAGE_REGISTRY=${image_registry}  \
+        IMAGE_REPO=${image_repo} \
+        push-handler
+)
+
+if [ "$branch" != $docs_source_branch ]; then
```
Comment (phoracek) [inline automation/publish.sh:25]: Could you change it to a positive check? The issue with this is that you make everything after this line "documentation script. If I add something to the bottom, it would be skipped everywhere except on the docs source branch.

File: docs/Dockerfile (hunk 13 above)
Comment (phoracek) [reply docs/Dockerfile:6]: You got me by the `make serve`. All good then.

File: docs/Dockerfile [hunk 19]
```diff
@@ -0,0 +1,6 @@
+ARG  BRANCH=master
+FROM quay.io/nmstate/kubernetes-nmstate-docs:${BRANCH}
+COPY Makefile Gemfile* /docs/
+RUN make -C docs install
+COPY . /docs/
```
Comment (phoracek) [inline docs/Dockerfile:5]: Why don't you copy everything into the container on line 3?

**Review by phoracek — CHANGES_REQUESTED** [review]

File: docs/Dockerfile (hunk 19 above)
Comment (qinqon) [reply docs/Dockerfile:5]: The idea is to have a pair of layers just with ruby deps and a pair of layers just with generated webpage.

File: automation/publish.sh (hunk 17 above)
Comment (qinqon) [reply automation/publish.sh:31]: In this case all the things are done in the Makefile so the image is kind of transparent, in fact is more of a CI thing.

We need the image to cache ruby deps, for Jekyll there are some of them that involve compile native code and takes some time.

File: docs/Dockerfile (hunk 19 above)
Comment (phoracek) [reply docs/Dockerfile:5]: Oh nice!

**Comment by phoracek** [comment]
/lgtm
/approve

Nice!
